src/spoke/BalanceSheet.sol

```diff
@@ -199,12 +199,12 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
     /// @inheritdoc IBalanceSheet
     function submitQueuedAssets(PoolId poolId, ShareClassId scId, AssetId assetId) external authOrManager(poolId) {
         AssetQueueAmount storage assetQueue = queuedAssets[poolId][scId][assetId];
-        if (assetQueue.deposits == 0 && assetQueue.withdrawals == 0) return;
```

Comment:
Why the removal of this early return? I understand the `assetCounter` helper variable usage as a result.

Reply:
Otherwise, we could not notify the hub that the snapshot value may have changed. i.e:
- Last hub snapshot is “synced”
- balanceSheet.issue()
- balanceSheet.submitQueuedAssets() //Nothing is sent here. Hub is not notified that the snapshot is no longer synced.

This is indeed a very border case, and probably the manager will use submitQueuedShares() instead. Nevertheless, from a user perspective, I would expect to trigger the snapshotHook for each submit* call